src/argilla/server/search_engine/base.py

```diff
@@ -69,54 +75,53 @@ def from_string(cls, metadata_property: MetadataProperty, string: str) -> "Metad
         return cls(metadata_property=metadata_property, values=string.split(","))
 
 
-# TODO: transform to `pydantic.BaseModel`
-@dataclasses.dataclass
-class IntegerMetadataFilter(MetadataFilter):
-    low: Optional[int] = None
-    high: Optional[int] = None
+NT = TypeVar("NT", int, float)
 
-    class RangeModel(BaseModel):
-        ge: Optional[int]
-        le: Optional[int]
 
-    def __post_init__(self):
-        if self.low is None and self.high is None:
-            raise ValueError("One of `low` or `high` value must be provided")
+class _RangeModel(GenericModel, Generic[NT]):
+    ge: Optional[NT]
+    le: Optional[NT]
 
-    @classmethod
-    def from_string(cls, metadata_property: MetadataProperty, string: str) -> "IntegerMetadataFilter":
-        model = cls.RangeModel.parse_raw(string)
-        return cls(metadata_property, low=model.ge, high=model.le)
 
+class NumericMetadataFilter(GenericModel, Generic[NT], MetadataFilter):
+    low: Optional[NT] = None
+    high: Optional[NT] = None
```

Comment:
maybe it's better to update this to `min` and `max` in order to be aligned with the settings we're saving in the DB